cmd/prom-aggregation-gateway/main.go

```diff
@@ -78,14 +79,26 @@ func mergeBuckets(a, b []*dto.Bucket) []*dto.Bucket {
 	return output
 }
 
+func makeTimestampMs() int64 {
+	return time.Now().UnixNano() / int64(time.Millisecond)
+}
+
 func mergeMetric(ty dto.MetricType, a, b *dto.Metric) *dto.Metric {
+	// Getting the metric timestamp or creating one when that metric is merged
+	// It will be used to cleanup old metrics that have not been merged lately
+	metricTimestamp := b.GetTimestampMs()
+	if metricTimestamp == 0 {
+		metricTimestamp = makeTimestampMs()
+	}
```

Comment:
Shouldn't this always set the latest timestamp, so we only remove things that haven't been touched recently? Or am I misunderstanding?

Reply:
At first, I thought the same, but then I researched a bit and these timestamps can be set by the emitter as well and used for other reasons. That's not common, though, probably why the original PAG logic just... dropped the value without caring much. I added that logic to check if it's set first just in case that we want to use timestamps in the future.